src/realm/sort_descriptor.hpp

```diff
@@ -115,6 +126,16 @@ class BaseDescriptor {
         struct ObjCache {
             ObjKey key;
             Mixed value;
+            std::optional<StringID> cached_string_id;
```

Comment:
Similarly, this is increasing the size of ObjCache from 24 bytes to 40 bytes to store data which could fix in `value`.

Reply:
OK, makes sense, but we need to store at least bool to flag whether the col is compressed or not.

Reply:
This might just turn into a giant mess, but we could add a `type_StringId` which is only used by Mixed which would let us store the string id in mixed with no overhead.